### ISSUE TITLE
Proposal: Consistent naming for filename and the contained class name

### ISSUE THREAD

**Issue by jslee02** [flexible-collision-library/fcl#162]
It seems the current convention for filenames is using underscore while the class name's convention is camel case. I suggest using one convention here and vote on using camelcase.

**Comment by sherm1** [comment]
I used consistent CamelCase in Simbody for classes and files as you are proposing, and I liked that. However, I have learned that many people feel strongly that only lower case should be used for file names, I think to avoid cross-platform and cross-network problems that mix case-sensitive and case-insensitive systems, and possibly for cleaner alphabetizing. [Google style guide](https://google.github.io/styleguide/cppguide.html#File_Names) uses MyClassName camel case but my_class_name.h file names. I've gotten used to that now and it seems fine as long as its done consistently.

**Comment by jslee02** [comment]
That sounds reasonable to me.

There are some files using upper case for abbreviations such as `BVH_` and `AABB`. I think then these should be changed to lowercase for consistency.

Also, we should consider using `_impl.h` rather than `-impl.h` for implementation files.

**Comment by sherm1** [comment]
> consider using _impl.h rather than -impl.h for implementation files

I actually prefer the hyphen there because it makes it (somewhat) clear that this is not part of the class name. So for class MyClassName we could have `my_class_name.h` and `my_class_name-impl.h` where the latter is seen as the same class name but with a suffix. There are actually a lot of real classes whose names end in `Impl` (often because of the PIMPL design pattern), and then we would have an ambiguity for a class named MyClassNameImpl: what is `my_class_name_impl.h`?

**Comment by sherm1** [comment]
I agree about downshifting the abbreviations in file names. Google style guide even suggests a standard way to deal with [acronyms in symbol names](https://google.github.io/styleguide/cppguide.html#Function_Names):
> Prefer to capitalize acronyms as single words (i.e. StartRpc(), not StartRPC()).

**Comment by jslee02** [comment]
I don't have a strong preference on suffix for implementation files so I'm fine with the hyphen.

> Google style guide even suggests a standard way to deal with acronyms in symbol names:
> 
> > Prefer to capitalize acronyms as single words (i.e. StartRpc(), not StartRPC()).

I also prefer to capitalize acronyms as single words to avoid confusion with the next word that start with a capital letter (e.g., prefer `HttpConnection` over `HTTPConnection`).
